evals/runners/_providers.py

```
@@ -0,0 +1,377 @@
+
+
+def get_pricing(model: str) -> dict[str, float]:
+    """Look up pricing for a model. Tries exact match, then prefix match
+    (handles dated snapshots like `gpt-4o-2024-11-20` mapping to `gpt-4o`).
+    Falls back to a conservative high estimate so the report never under-reports.
+    """
+    if model in MODEL_PRICING:
+        return MODEL_PRICING[model]
+    # Try prefix match — dated snapshots map to their alias.
+    for alias, pricing in MODEL_PRICING.items():
+        if model.startswith(alias + "-") or model.startswith(alias + "@"):
```

Ruff (0.15.14)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)